composable_kernel/include/problem_transform/transform_forward_convolution3d_into_gemm_v4r4r4_ndhwc_kzyxc_ndhwk.hpp

```diff
@@ -0,0 +1,306 @@
+
+    return make_tuple(in_gemmk0_gemmm_gemmk1_grid_desc,
+                      wei_gemmk0_gemmn_gemmk1_grid_desc,
+                      out_gemmm_gemmn_grid_desc);
+    // n0_gemmk0_gemmn_gemmk1, gemmk0_gemmn_gemmk1, n0_gemmm_gemmn
+}
+
+template <typename... In,
+          typename... Wei,
+          typename... Out,
+          typename ConvStrides,
+          typename ConvDilations,
+          typename InLeftPads,
+          typename InRightPads,
+          index_t GemmK1Value>
+__host__ __device__ constexpr auto
+transform_forward_convolution3d_into_gemm_v4r4r4_nhwc_kyxc_nhwk_pad_split_batch(
+    const TensorDescriptor<In...>& in_n_di_hi_wi_c_grid_desc,
+    const TensorDescriptor<Wei...>& wei_k_z_y_x_c_grid_desc,
+    const TensorDescriptor<Out...>& out_n_do_ho_wo_k_grid_desc,
+    const ConvStrides& conv_strides,
+    const ConvDilations& conv_dilations,
+    const InLeftPads& in_left_pads,
+    const InRightPads& in_right_pads,
+    Number<GemmK1Value>)
+{
+    constexpr auto I0 = Number<0>{};
+    constexpr auto I1 = Number<1>{};
+    constexpr auto I2 = Number<2>{};
+    constexpr auto I3 = Number<3>{};
+    constexpr auto I4 = Number<4>{};
+
+    constexpr auto GemmK1 = Number<GemmK1Value>{};
+
+    const auto N = in_n_di_hi_wi_c_grid_desc.GetLength(I0);
+    const auto C = in_n_di_hi_wi_c_grid_desc.GetLength(I4);
+    const auto K = out_n_do_ho_wo_k_grid_desc.GetLength(I4);
+
+    const auto Di = in_n_di_hi_wi_c_grid_desc.GetLength(I1);
+    const auto Hi = in_n_di_hi_wi_c_grid_desc.GetLength(I2);
+    const auto Wi = in_n_di_hi_wi_c_grid_desc.GetLength(I3);
+
+    const auto Do = out_n_do_ho_wo_k_grid_desc.GetLength(I1);
+    const auto Ho = out_n_do_ho_wo_k_grid_desc.GetLength(I2);
+    const auto Wo = out_n_do_ho_wo_k_grid_desc.GetLength(I3);
+
+    const auto Z = wei_k_z_y_x_c_grid_desc.GetLength(I1);
+    const auto Y = wei_k_z_y_x_c_grid_desc.GetLength(I2);
+    const auto X = wei_k_z_y_x_c_grid_desc.GetLength(I3);
+
+    const auto ConvStrideD = conv_strides[I0];
+    const auto ConvStrideH = conv_strides[I1];
+    const auto ConvStrideW = conv_strides[I2];
+
+    const auto ConvDilationD = conv_dilations[I0];
+    const auto ConvDilationH = conv_dilations[I1];
+    const auto ConvDilationW = conv_dilations[I2];
+
+    const auto InLeftPadD = in_left_pads[I0];
+    const auto InLeftPadH = in_left_pads[I1];
+    const auto InLeftPadW = in_left_pads[I2];
+
+    const auto InRightPadD = in_right_pads[I0];
+    const auto InRightPadH = in_right_pads[I1];
+    const auto InRightPadW = in_right_pads[I2];
+
+    // N1 should satisfy that
+    //   1) N1 = 2^i;
+    //   2) N1 * (Do * Ho * Wo * K) < (2^31 - 1)
+    //   3) N1 * (Di * Hi * Wi * C) < (2^31 - 1)
+    //
+    // Do NOT confuse (B, N1) in this function with (B, N1) in gridewise GEMM.
+    auto N1 = N + 1;
+
+    {
+        const auto stride =
+            std::max(long_index_t(Do) * Ho * Wo * K, long_index_t(Di) * Hi * Wi * C);
```

Comment:
use math::max instead of std::max.

This function may be called on GPU one day